doc/OnlineDocs/modeling_extensions/gdp/modeling.rst

```diff
@@ -0,0 +1,415 @@
+
+.. doctest::
+
+    >>> m.p = LogicalConstraint(expr=m.Y[1].implies(m.Y[2] & m.Y[3]) | m.Y[4])
+    >>> m.p.pprint()
+    p : Size=1, Index=None, Active=True
+        Key  : Body                         : Active
+        None : Y[1] >> (Y[2] & Y[3]) | Y[4] :   True
+
+Supported Logical Operators
+---------------------------
+
+Pyomo.GDP logical expression system supported operators and their usage are listed in the table below.
+
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Operator     | Infix                  | Method                            | Function                       |
++==============+========================+===================================+================================+
+| Conjunction  | :code:`Y[1] & Y[2]`    | :code:`Y[1].and_(Y[2])`           | :code:`And(Y[1],Y[2])`         |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Disjunction  | :code:`Y[1] | Y[2]`    | :code:`Y[1].or_(Y[2])`            | :code:`Or(Y[1],Y[2])`          |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Negation     | :code:`~Y[1]`          |                                   | :code:`Not(Y[1])`              |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Exclusive OR | :code:`Y[1] ^ Y[2]`    | :code:`Y[1].xor(Y[2])`            | :code:`Xor(Y[1], Y[2])`        |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Implication  | | :code:`Y[1] >> Y[2]` | :code:`Y[1].implies(Y[2])`        | :code:`Implies(Y[1], Y[2])`    |
+|              | | :code:`Y[2] << Y[1]` |                                   |                                |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Equivalence  | :code:`Y[1] == Y[2]`   | :code:`Y[1].equivalent_to(Y[2])`  | :code:`Equivalent(Y[1], Y[2])` |
++--------------+------------------------+-----------------------------------+--------------------------------+
+
+In addition, the following constraint-programming-inspired operators are provided: ``Exactly``, ``AtMost``, and ``AtLeast``.
+These predicates enforce, respectively, that exactly, at most, or at least N of their ``BooleanVar`` arguments are ``True``.
+
+Usage:
+
+- :code:`AtLeast(3, Y[1], Y[2], Y[3])`
+- :code:`AtMost(3, Y)`
+- :code:`Exactly(3, Y)`
+
+.. doctest::
+
+    >>> m = ConcreteModel()
+    >>> m.my_set = RangeSet(4)
+    >>> m.Y = BooleanVar(m.my_set)
+    >>> m.p = LogicalConstraint(expr=AtLeast(3, m.Y))
+    >>> TransformationFactory('core.logical_to_linear').apply_to(m)
+    >>> m.logic_to_linear.pprint()  # constraint auto-generated by transformation
```

Comment:
I was trying to keep things relatively straightforward on the first pass. I'll change it to be a block, since it would more easily facilitate future expansion.